packages/xod-pm/src/fetch.js

```diff
@@ -87,12 +95,39 @@ export const fetchLibXodball = R.curry((swaggerUrl, libQuery) =>
                   request: libQuery,
                   params,
                 }),
-                () => fetchLibXodball(swaggerUrl, libQuery)
+                () => fetchOneLibrary(swaggerUrl, libQuery)
               )
             )
             .then(R.prop('obj'))
+            .then(xodball =>
+              R.compose(eitherToPromise, fromXodballData)(xodball)
+                .catch(rejectWithCode(ERR_CODES.INVALID_XODBALL))
+            )
         ),
         parseLibQuery
       )(libQuery)
     )
 );
+
+// :: URL -> LibName -> Promise [Project] Error
+export const fetchLibrary = R.curry((swaggerUrl, libQuery) => {
+  // :: [LibName] -> [Project] -> [LibName] -> Promise [Project] Error
+  const fetchReqursively = R.curry((fetchedLibNames, fetchedLibs, libNamesToFetch) => {
```

Comment:
Probably, instead of storing libs and lib names separately, it would be more convenient to store them in an object with lib names as keys.

Reply:
Nice idea! 👍